### ISSUE TITLE
redim.range doesn't work for datetimes on latest dev

### ISSUE THREAD

**Issue by ahuang11** [ioam/holoviews#3044]
```
tsz = pd.DataFrame([0, 1, 2], index=pd.date_range('2017-02-01', end='2017-02-03')).rename_axis('x')
tsz.hvplot().redim.range(x=('2017-01-01', '2017-02-03'))

tsz = pd.DataFrame([0, 1, 2], index=pd.date_range('2017-02-01', end='2017-02-03')).rename_axis('x')
tsz.hvplot().redim.range(x=(pd.to_datetime('2017-01-01'), pd.to_datetime('2017-02-03')))
```

```
/mnt/c/Users/user/GOOGLE~1/Bash/holoviews/holoviews/plotting/bokeh/element.py in initialize_plot(self, ranges, plot, plots, source)
    722             element = self.hmap.last
    723         key = util.wrap_tuple(self.hmap.last_key)
--> 724         ranges = self.compute_ranges(self.hmap, key, ranges)
    725         self.current_ranges = ranges
    726         self.current_frame = element

/mnt/c/Users/user/GOOGLE~1/Bash/holoviews/holoviews/plotting/plot.py in compute_ranges(self, obj, key, ranges)
    381             if (not (axiswise and not isinstance(obj, HoloMap)) or
    382                 (not framewise and isinstance(obj, HoloMap))):
--> 383                 self._compute_group_range(group, elements, ranges)
    384         self.ranges.update(ranges)
    385         return ranges

/mnt/c/Users/user/GOOGLE~1/Bash/holoviews/holoviews/plotting/plot.py in _compute_group_range(group, elements, ranges)
    454             data_range = util.max_range(values['data'])
    455             combined = util.dimension_range(data_range[0], data_range[1],
--> 456                                             hard_range, soft_range)
    457             dranges = {'data': data_range, 'hard': hard_range,
    458                        'soft': soft_range, 'combined': combined}

/mnt/c/Users/user/GOOGLE~1/Bash/holoviews/holoviews/core/util.py in dimension_range(lower, upper, hard_range, soft_range, padding, log)
    883     lower, upper = max_range([(lower, upper), soft_range])
    884     dmin, dmax = hard_range
--> 885     lower = lower if dmin is None or not np.isfinite(dmin) else dmin
    886     upper = upper if dmax is None or not np.isfinite(dmax) else dmax
    887     return lower, upper

TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

**Comment by philippjfr** [comment]
Thanks for the report. This should be an easy fix, ``np.isfinite`` simply needs to be ``isfinite`` since we already defined a general ``isfinite`` function in the same module.

**Comment by philippjfr** [comment]
Fixed in #3045
